src/lab16.py

```diff
@@ -345,10 +345,11 @@ def __init__(self, node, parent, previous, frame):
             y_dependencies = [self.previous.y, self.previous.height]
         else:
             y_dependencies = [self.parent.y]
-        self.y = ProtectedField(self, "y", self.parent, y_dependencies)
+        self.y = ProtectedField(
+            self, "y", self.parent, y_dependencies)
 
         self.children = ProtectedField(self, "children", self.parent, None,
-            [self.height])
+            [])
```

Comment:
This is the interesting change, right? Come to think of it, I'm not sure why this was ever here, or even why we have a frozen_invalidations field.

Reply:
Yes this is the interesting one.